packages/ag-charts-community/src/scene/clip.ts

```diff
@@ -0,0 +1,52 @@
+import { jsonDiff } from '../util/json';
+import { BBox } from './bbox';
+import { Node, RedrawType, type RenderContext } from './node';
+
+type Constructor<T> = new (...args: any[]) => T;
+type Circle = { x: number; y: number; radius: number };
+
+export type ClippableOutsideType<T> = T & {
+    clipShape?: BBox | Circle;
+    setClipMask: (shape?: BBox | Circle) => void;
+};
+
+export function ClippableOutside<N extends Node>(Parent: Constructor<N>): Constructor<ClippableOutsideType<N>> {
+    const ParentNode = Parent as Constructor<Node>;
+    class ClippableOutsideInternal extends ParentNode {
+        clipShape?: BBox | Circle;
+
+        setClipMask(shape?: BBox | Circle) {
+            if (jsonDiff(this.clipShape, shape) != null) {
+                this.markDirty(this, RedrawType.MINOR);
+            }
+
+            this.clipShape = shape;
+        }
+
+        override postRender(renderCtx: RenderContext) {
+            const { ctx } = renderCtx;
+            const { clipShape } = this;
+
+            if (!clipShape) {
+                super.postRender(renderCtx);
+                return;
+            }
+
+            const prevOperation = ctx.globalCompositeOperation;
```

Comment:
Try `ctx.save()` here, and `ctx.restore()` below - these are more bullet-proof ways of restoring `ctx` state, and what we are using elsewhere?

Reply:
Oh of course, thanks!